File: MODBackend/AuthenticateService/Models/Skill.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace AuthenticateService.Models
{
    [Table("Skill")]
    public class Skill
    {
        [Key]
        public string SkillId { get; set; }

        public string SkillName { get; set; }

        public string toc { get; set; }
        public string prerequisites { get; set; }

        public Nullable<int> fees { get; set; }
        public int duration { get; set; }

        public IEnumerable<Training> Trainings { get; set; }
    }
}

File: MODUI/src/app/Models/skill.ts
export class Skill {
    public skillId: string;
      
    public skillName : string;

    public toc : string;

    public prerequisites : string;

    public fees : number;
    public  duration : number;


    constructor() { 
        this.skillId="S" + Math.round((Math.random()*9000));
        
        
    }   
}
